app/src/main/java/com/duckduckgo/app/browser/BrowserTrackersAnimatorHelper.kt

```diff
@@ -20,121 +20,48 @@ import android.animation.AnimatorSet
 import android.animation.ObjectAnimator
 import android.annotation.SuppressLint
 import android.app.Activity
-import android.content.res.Resources
 import android.graphics.drawable.AnimatedVectorDrawable
 import android.view.View
 import android.widget.ImageView
-import android.widget.TextView
 import androidx.constraintlayout.widget.ConstraintLayout
 import androidx.constraintlayout.widget.ConstraintSet
 import com.duckduckgo.app.cta.ui.Cta
 import com.duckduckgo.app.cta.ui.DaxDialogCta
 import com.duckduckgo.app.privacy.renderer.TrackersRenderer
 import com.duckduckgo.app.trackerdetection.model.TrackingEvent
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.Job
-import kotlinx.coroutines.SupervisorJob
-import kotlinx.coroutines.delay
-import kotlinx.coroutines.launch
-import javax.inject.Inject
-import kotlin.coroutines.CoroutineContext
 
-class BrowserTrackersAnimatorHelper @Inject constructor() : CoroutineScope {
-
-    override val coroutineContext: CoroutineContext
-        get() = SupervisorJob() + Dispatchers.Main
+class BrowserTrackersAnimatorHelper {
 
     private var logosStayOnScreenDuration: Long = TRACKER_LOGOS_DELAY_ON_SCREEN
-
-    private var loadingAnimation: AnimatorSet = AnimatorSet()
     private var trackersAnimation: AnimatorSet = AnimatorSet()
-    private var typingAnimationJob: Job? = null
-
-    private fun createScanningAnimation(resources: Resources, loadingText: TextView): Job {
-        return launch {
-            while (true) {
-                if (loadingText.text.contains("...")) {
-                    loadingText.text = resources.getString(R.string.trackersAnimationText)
-                } else {
-                    loadingText.text = resources.getString(R.string.trackersAnimationDotText, loadingText.text)
-                }
-                delay(SCANNING_DELAY)
-            }
-        }
-    }
 
-    fun createLoadedAnimation(
-        cta: Cta?,
-        activity: Activity,
-        container: ConstraintLayout,
-        loadingText: View,
-        views: List<View>,
-        events: List<TrackingEvent>?
-    ) {
-        if (loadingAnimation.isRunning) {
-            loadingAnimation.end()
-        }
+    fun startTrackersAnimation(cta: Cta?, activity: Activity, container: ConstraintLayout, views: List<View>, events: List<TrackingEvent>?) {
```

Comment:
NIT: remove nullability from `events: List<TrackingEvent>?` (and in all the methods within this class) and only check if it's empty here.

Reply:
Done!